runtime/extern.c

```diff
@@ -278,13 +274,6 @@ static intnat extern_output_length(void)
 
 /* Exception raising, with cleanup */
 
-static void extern_out_of_memory(void)
-{
-  extern_replay_trail();
-  free_extern_output();
-  caml_raise_out_of_memory();
-}
-
```

Comment:
This change is not correct. If you're in "raise OOM" mode, then you need to replay the trail and free the output before you can raise the exception.